### PR TITLE
:sparkles: add ability to extract sub-images from PDFs

### PR DESCRIPTION
**Pull request by ianardee**
## Description
Add the ability to extract sub-images from individual pages.

Auto-detect if the source image is a PDF when constructing the image extractor.
If it's a PDF, extract each PDF page to an individual image.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires a change to the official Guide documentation.

### CODE REVIEW

File: src/main/java/com/mindee/extraction/ImageExtractor.java [hunk 1]
```diff
@@ -52,14 +74,67 @@ public <FieldT extends PositionData> List<ExtractedImage> extractImages(List<Fie
...
+   * @param outputName The base output filename, must have an image extension.
    * @return A list of {@link ExtractedImage}.
    */
-  public <FieldT extends PositionData> List<ExtractedImage> extractImages(List<FieldT> fields, String filename) {
+  public <FieldT extends PositionData> List<ExtractedImage> extractImages(List<FieldT> fields, String outputName) {
+    if (this.getPageCount() > 1) {
+      throw new MindeeException("Input file has more than one page, use the `extractPageImages` method instead.");
+    }
+    return extractFromPage(fields, 0, outputName);
+  }
```
Comment (sebastianMindee) [inline src/main/java/com/mindee/extraction/ImageExtractor.java:87]: Why not just use the same function for both? Separating single vs multi-page seems like an easy way to get things mixed-up during usage, i.e. if you're treating batches of file

File: src/main/java/com/mindee/extraction/ImageExtractor.java [hunk 2]
```diff
@@ -52,14 +74,67 @@ public <FieldT extends PositionData> List<ExtractedImage> extractImages(List<Fie
 
   /**
    * Extract images from a list of fields having position data.
+   * Use this when the input file is an image or a single-page PDF.
    * @param fields List of Fields to extract.
-   * @param filename The base output filename.
+   * @param outputName The base output filename, must have an image extension.
```
Comment (sebastianMindee) [inline src/main/java/com/mindee/extraction/ImageExtractor.java:79]: Add a unit test for this

**Review by sebastianMindee — CHANGES_REQUESTED** [review]
Missing one minor UT, otherwise OK

File: src/main/java/com/mindee/extraction/ImageExtractor.java (hunk 1 above)
Comment (ianardee) [reply src/main/java/com/mindee/extraction/ImageExtractor.java:87]: right you are, done.

File: src/main/java/com/mindee/extraction/ImageExtractor.java (hunk 2 above)
Comment (ianardee) [reply src/main/java/com/mindee/extraction/ImageExtractor.java:79]: test added

**Review by sebastianMindee — APPROVED** [review]

**Pull request merged**